### PR TITLE
config: Add initial sysbuild files

### PR DESCRIPTION
**Pull request by nordicjm**
Adds initial WIP sysbuild files

> !!!!!!!!!! Please delete the instructions below and replace with PR description
>
> If you have an issue number, please use a syntax of
> `Fixes #12345` and a brief change description
>
> If you do not have an issue number, please have a good description of
> the problem and the fix. Help the reviewer understand what to expect.
>
> Make sure you delete these instructions (to prove you have read them).
>
> !!!!!!!!!! Instructions end

### CODE REVIEW

File: config/nrfconnect/chip-module/CMakeLists.txt [hunk 1]
```diff
@@ -244,7 +244,7 @@ endif()
 # Define 'factory_data' target for generating a factory data partition
 # ==============================================================================
 
-if (CONFIG_CHIP_FACTORY_DATA_BUILD)
+if(CONFIG_CHIP_FACTORY_DATA_BUILD AND NOT SYSBUILD)
```
Comment (markaj-nordic) [inline config/nrfconnect/chip-module/CMakeLists.txt:247]: Why skipping factory data when using sysbuild? Is it because of WIP?

Comment (nordicjm) [reply config/nrfconnect/chip-module/CMakeLists.txt:247]: sysbuild code is in the other file

Comment (markaj-nordic) [reply config/nrfconnect/chip-module/CMakeLists.txt:247]: Ok, I see the new file is used in sysbuild CMake.

File: config/nrfconnect/chip-module/generate_factory_data_sysbuild.cmake [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+  # Create a .hex file with factory data in CBOR format based on the JSON file created previously
+  nrfconnect_create_factory_data(factory_data
+                                 ${FACTORY_DATA_SCRIPT_PATH}
+                                 ${FACTORY_DATA_SCHEMA_PATH}
+                                 ${OUTPUT_FILE_PATH}
+                                 ${DEFAULT_IMAGE}
+  )
+
+  if(SB_CONFIG_CHIP_FACTORY_DATA_MERGE_WITH_FIRMWARE)
```
Comment (ArekBalysNordic) [inline config/nrfconnect/chip-module/generate_factory_data_sysbuild.cmake:219]:
```suggestion
  if(SB_CONFIG_MATTER_FACTORY_DATA_MERGE_WITH_FIRMWARE)
```

**Review by ArekBalysNordic — CHANGES_REQUESTED** [review]

File: config/nrfconnect/chip-module/generate_factory_data_sysbuild.cmake (hunk 2 above)
Comment (nordicjm) [reply config/nrfconnect/chip-module/generate_factory_data_sysbuild.cmake]: Fixed

**Review by ArekBalysNordic — APPROVED** [review]

**Review by markaj-nordic — APPROVED** [review]

**Pull request merged**